### PR TITLE
expression: implement `elt` function pushdown

### LINKED ISSUE #36734: Implement elt function push down to TiFlash
## Enhancement

support elt function push down to TiFlash.

For details: [pingcap/tiflash#5104](https://github.com/pingcap/tiflash/issues/5104)

### PR DESCRIPTION
**Pull request by Willendless**
### What problem does this PR solve?

Issue Number: close #36734 
related issue: [pingcap/tiflash#5104](https://github.com/pingcap/tiflash/issues/5104)

Problem Summary:

Support push down `elt` function to TiFlash

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Support to pushdown elt to TiFlash
```

### CODE REVIEW

File: planner/core/integration_test.go [hunk 1]
```diff
@@ -7224,3 +7224,38 @@ func TestIssue36609(t *testing.T) {
...
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t")
+	tk.MustExec("create table t(a int, b varchar(20))")
+	tk.MustExec("insert into t values(2147483647, '32')")
+	tk.MustExec("insert into t values(12, 'abc')")
+	tk.MustExec("set @@tidb_allow_mpp=1; set @@tidb_enforce_mpp=1")
+	tk.MustExec("set @@tidb_isolation_read_engines = 'tiflash'")
+
+	// Create virtual tiflash replica info.
+	dom := domain.GetDomain(tk.Session())
```
Comment (SeaRise) [inline planner/core/integration_test.go:7241]: use `store, dom, clean := testkit.CreateMockStoreAndDomain(t)` to create `dom`.

**Review by SeaRise — APPROVED** [review]
Others LGTM

**Review by windtalker — APPROVED** [review]
LGTM

File: planner/core/integration_test.go (hunk 1 above)
Comment (Willendless) [reply planner/core/integration_test.go:7241]: Sure

**Pull request merged**